actions/class.Tests.php

```diff
@@ -167,7 +168,14 @@ public function delete()
             $lockManager->releaseLock($instance, $userId);
         }
 
-        if ($this->service->deleteTest($instance)) {
+        // If it is a class deletion
+        if ($uri === $classUri) {
```

Comment:
Not necessary in this case. I would also recommend to extract this if/else conditions to a new method with early returns. Boy Scout Rule: https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html

Wdyt?

```suggestion
        if ($uri === $classUri) {
```